pkg/media/media_data_parser.go

```diff
@@ -113,15 +113,22 @@ func ParseSegmentMediaData(ctx context.Context, mp4bs []byte) (*model.SegmentMed
 	if err != nil {
 		return nil, fmt.Errorf("error creating SegmentMetadata pipeline: %w", err)
 	}
-	demux.Connect("pad-added", onPadAdded)
+	_, err = demux.Connect("pad-added", onPadAdded)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting pad-add: %w", err)
+	}
 
 	go func() {
-		HandleBusMessages(ctx, pipeline)
+		if err := HandleBusMessages(ctx, pipeline); err != nil {
+			log.Log(ctx, "error handling bus messages", "error", err)
```

Comment:
Nit: this is slightly incorrect - an error returning from `HandleBusMessages` usually indicates the pipeline itself exiting with an error rather than the bus handler. These could be as simple as 
```suggestion
			log.Error(ctx, "pipeline error", "error", err)
```